### DISCUSSION TITLE
What does the push option mean exactly?

### DISCUSSION THREAD

**Discussion by GersonTrj** [docker/build-push-action#1168] [Q&A] [answered]
What are the commands ran when you set `push: false` and `push: true` and what do they mean exactly? 

if I set push to false, the action will build the dockerfile without pushing to the container registry? (that'd be great for testing).

**Comment by crazy-max** [comment, marked as answer]
`push` is the same as `--push` flag of build command: https://docs.docker.com/reference/cli/docker/buildx/build/#push. `true` will push to the registry while false does not set `--push` for the build command in the action which would result with nothing being exported.

**Reply by GersonTrj** [reply]
so essentially with push false, it just builds the image right? so it could be used for looking for errors in the app build without pushing it to the registry.

**Reply by crazy-max** [reply]
Yes
